### DISCUSSION TITLE
Static input file format inhibits flexibility

### DISCUSSION THREAD

**Discussion by rafmudaf** [OpenFAST/openfast#732] [Ideas]
The current input handling in the NWTC Library expects a mostly static input file format (I say "mostly" because at least one example of a dynamic input file format exists in `beamdyn_driver`). This means that any new feature which needs input will cause a breaking API change and, therefore, requires a new major version number according to the rules of [semantic versioning](https://semver.org).

It has been proposed to implement a more dynamic input system with a flexible syntax and default values where appropriate (something dictionary based, yaml, json, etc). This would allow backwards compatibility when new features are introduced and enable simpler restructuring of the input file.

**Comment by gantech** [comment]
Great. I totally support this idea.

**Comment by jjonkman** [comment]
While I am generally supportive of the proposal, we chose against it in the past due the large amount of work involved.  Not only does the input-file processing within OpenFAST have to change, but all of the scripts that run batch simulations of OpenFAST must change.  Of course, such an endeavor does not have to done at one time, but can be tacked on a module-by-module basis.
